src/nock.utils.test.ts

```diff
@@ -471,16 +472,151 @@ describe('nock utils', () => {
         });
       });
     });
+
+    describe('when sensitive information are URL encoded', () => {
+      it('replaces api key in body', async () => {
+        const secret = 'шеллы';
+        const encodedSecret = encodeURIComponent(secret);
+        const definition: RecordingDefinition = {
+          scope: BASE_URL,
+          path: '/get?text=123',
+          method: 'GET',
+          status: 200,
+          body: {
+            my_api_key: encodedSecret,
+          },
+        };
+
+        replaceCredentialInDefinition({
+          definition,
+          scheme: {
+            id: 'api-key',
+            type: SecurityType.APIKEY,
+            in: ApiKeyPlacement.BODY,
+            name: 'api_key',
+          },
+          baseUrl: BASE_URL,
+          credential: secret,
+          placeholder: TMP_PLACEHOLDER,
+        });
+
+        expect(definition).toEqual({
+          scope: BASE_URL,
+          path: '/get?text=123',
+          method: 'GET',
+          status: 200,
+          body: {
+            my_api_key: TMP_PLACEHOLDER,
+          },
+        });
+      });
+
+      it('replaces placeholder for api key in body', async () => {
```

Comment:
Do I understand it right that `replaceCredentialInDefinition` is used also to replace place holder with credential when running tests (without recording)?

Reply:
Yes, should I find better naming for this?